src/main/host/thread_preload.c

```diff
@@ -223,6 +228,7 @@ static inline void _threadpreload_waitForNextEvent(ThreadPreload* thread) {
     MAGIC_ASSERT(_threadPreloadToThread(thread));
     utility_assert(thread->ipc_data);
     shimevent_recvEventFromPlugin(thread->ipc_data, &thread->currentEvent);
+    host_lockShimShmemLock(thread->base.host);
```

Comment:
Maybe add comments here about where this is expected to be unlocked since there's no nearby `host_unlockShimShmemLock()` call.

Reply:
Done